naga/src/front/wgsl/lower/mod.rs

```diff
@@ -1065,6 +1069,29 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
             match decl.kind {
                 ast::GlobalDeclKind::Fn(ref f) => {
                     let lowered_decl = self.function(f, span, &mut ctx)?;
+                    if !f.comments.is_empty() {
+                        match lowered_decl {
+                            LoweredGlobalDecl::Function { handle, .. } => {
+                                ctx.module
+                                    .get_comments_or_insert_default()
+                                    .functions
+                                    .insert(
+                                        handle,
+                                        f.comments.iter().map(|s| s.to_string()).collect(),
+                                    );
+                            }
+                            LoweredGlobalDecl::EntryPoint(index) => {
+                                ctx.module
+                                    .get_comments_or_insert_default()
+                                    .entry_points
+                                    .insert(
+                                        index,
+                                        f.comments.iter().map(|s| s.to_string()).collect(),
```

Comment:
I'm not sure how the doc comments will be consumed but wouldn't it make sense if they didn't contain the `///`, `//!` (and so on) inside? We could then concatenate them so that each item has a single `String` as its doc comment.

Reply:
I have no strong feelings about it, but I took the most "conservative" and unopiniated approach.

Some comments look like 
```rs
////////
// SOME HEADING
////////
```

or sometimes ascii art is in comments, and messing up with what user wrote may be surprising, so I kept it.

Oh and also, there are edge cases between `// comment` and `//comment` which could lead to unwanted space at the beginning, or we should take the most indented... cases add up fast and start to be opinionated quickly.

Reply:
Ok, we can change this later if needed.